channel/channel.go

```diff
@@ -327,17 +325,20 @@ func (c *Channel) SignAndAddState(s state.State, sk *[]byte) (state.SignedState,
 
 // UpdateWithChainEvent mutates the receiver if provided with a "new" chain event (with a greater block number than previously seen)
 func (c *Channel) UpdateWithChainEvent(event chainservice.Event) (*Channel, error) {
-	if event.BlockNum() < c.latestBlockNumber {
+	if event.BlockNum() < c.OnChain.LatestBlockNumber {
 		return c, nil // ignore stale information TODO: is this reorg safe?
 	}
-	c.latestBlockNumber = event.BlockNum()
+	c.OnChain.LatestBlockNumber = event.BlockNum()
 	switch e := event.(type) {
 	case chainservice.AllocationUpdatedEvent:
-		c.OnChainFunding[e.AssetAddress] = e.AssetAmount
+		c.OnChain.Holdings[e.AssetAddress] = e.AssetAmount
+		// TODO: update OnChain.StateHash and OnChain.Outcome
 	case chainservice.DepositedEvent:
-		c.OnChainFunding[e.Asset] = e.NowHeld
+		c.OnChain.Holdings[e.Asset] = e.NowHeld
```

Comment:
Outside the scope of this PR but having the `engine` reach into the channel struct and manually update the holdings seems to indicate a tight coupling between the engine and the channel implementation. Might be worth the channel struct exposing an `UpdatingHoldings` API, so the engine behaviour is unaffected by the implementation of the channel struct.